### PR TITLE
dns+ruler: Improve tests, block scheme for now.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: cmd/thanos/rule.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func registerRule(m map[string]setupFunc, app *kingpin.Application, name string)
 
 	objStoreConfig := regCommonObjStoreFlags(cmd, "", false)
 
-	queries := cmd.Flag("query", "Addresses of statically configured query API servers (repeatable). The scheme may be prefixed with 'dns+' or 'dnssrv+' to detect query API servers through respective DNS lookups.").
+	queries := cmd.Flag("query", "Addresses of statically configured query API servers (repeatable). The scheme may be prefixed with 'dns+' or 'dnssrv+' to detect query API servers through respective DNS lookups. No scheme prefix allowed (https://github.com/improbable-eng/thanos/issues/1025)").
```
Comment (povilasv) [inline cmd/thanos/rule.go]: I think the comment here doesn't make sense. As we have a sentence `The scheme may be prefixed with 'dns+' or 'dnssrv+'` 
and after that a sentence `No scheme prefix allowed (https://github.com/improbable-eng/thanos/issues/1025)")`

File: cmd/thanos/rule.go [hunk 2]
```diff
@@ -130,6 +130,13 @@ func registerRule(m map[string]setupFunc, app *kingpin.Application, name string)
 
 		lookupQueries := map[string]struct{}{}
 		for _, q := range *queries {
+			if q == "" {
+				return errors.New("Static querier address cannot be empty")
+			}
+			if strings.Contains(q, "://") {
```
Comment (povilasv) [inline cmd/thanos/rule.go]: So we won't support `dns://` urls anymore?

File: cmd/thanos/rule.go (hunk 1 above)
Comment (bwplotka) [reply cmd/thanos/rule.go]: yes, We need to reword ;p

It's tricky as we can do `dns+abc:port`, `abc:port` etc, but we cannot put `dns+http://abc:port` or http://abc:port` due to implementation limitations

File: cmd/thanos/rule.go (hunk 2 above)
Comment (bwplotka) [reply cmd/thanos/rule.go]: We never was suporting those. Only `dns+` URLS and others with `+`

**Pull request closed without merging**